lib/model/course/course.dart

```diff
@@ -1,27 +1,310 @@
 class Course {
   final String name;
   final String faculty;
-  final String lectureTime;
+  final List<Lecture> lecturesPerWeek;
   final String description;
   final int hoursPerWeek;
   final int ects;
   final String professorEmail;
   final String professorName;
-  final int availableForStudent;
+  final CourseAvailability available;
   bool isFavourite;
-  final int availability;
 
-  Course(this.name, this.faculty, this.lectureTime, this.description,
+  Course(this.name, this.faculty, this.lecturesPerWeek, this.description,
       this.hoursPerWeek, this.ects, this.professorEmail, this.professorName,
-      this.availableForStudent, this.isFavourite, this.availability);
+      this.available, this.isFavourite) {
+    //Set this course as parent of every lectures contained
+    //Required for timetable
+    lecturesPerWeek.forEach((l) => l.course = this);
+  }
+
+  bool occursOnDay(Weekday weekDay) {
+    //Is there a lecture on searched day
+    return lecturesPerWeek.any((lecture) => lecture.weekday == weekDay);
+  }
 }
 
+class Lecture {
+  final Campus campus;
+  final Weekday weekday;
+  final DayTime startDayTime;
+  final DayTime endDayTime;
+  final String room;
+  Course course;
+
+  Lecture(this.campus, this.weekday, this.startDayTime, this.endDayTime,
+      this.room);
+
+  //Return int which helps to get the order of lectures in week correct
+  int sortValue() {
+    // result looks like: whhmm
+    return WeekdayUtility.getWeekdayAsInt(weekday) * 10000 +
+        startDayTime.hour * 100 + startDayTime.minute;
+  }
+}
+
+class DayTime {
+  final int hour;
+  final int minute;
+
+  DayTime(this.hour, this.minute);
+
+  @override
+  String toString() {
+    String hourString = hour.toString();
+    if (hour <= 9)
+      hourString = "0" + hourString;
+
+    String minuteString = minute.toString();
+    if (minute <= 9)
+      minuteString = "0" + minuteString;
+
+    return '$hourString:$minuteString';
+  }
+}
+
+//Todo: This needs to be replaced with enum style of implementation in future see below
 class CourseDefinitions {
-  static const int AVAILABILITY_AVAILABLE = 0;
-  static const int AVAILABILITY_PENDING = 1;
-  static const int AVAILABILITY_UNAVAILABLE = 2;
   static const List<String> DEPARTMENTS = [
     "01", "02", "03", "04", "05", "06", "07", "08", "09", "10", "11", "12",
     "13", "14"
   ];
-}
\ No newline at end of file
+}
+
+enum CourseAvailability { AVAILABLE, PENDING, UNAVAILABLE }
+enum Weekday { Mon, Tue, Wed, The, Fri, Sat, Sun }
+enum Campus { KARLSTRASSE, LOTHSTRASSE, PASING }
+//enum Faculty {
```

Comment:
Just for documentation, those comments will stay. We will rebuild this class later based on those things.